extra/puppet_sign.rb

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env ruby
+#
+# This takes a base64 encoded CSR as input and uploads it to foreman-proxy for analysis.
+# It is used to verify if the CSR should be signed by PuppetCA.
+#
+
+require 'net/http'
+require 'net/https'
+require 'fileutils'
+require 'json'
+require 'yaml'
+require 'socket'
+
+def log(message)
+  puts message
+  `logger "[PuppetCA CSR Verify] #{message}"`
+end
+
+csr = STDIN.read
+
+if RbConfig::CONFIG['host_os'] =~ /freebsd|dragonfly/i
+  settings_file = '/usr/local/etc/foreman-proxy/settings.yml'
+else
+  settings_file = '/etc/foreman-proxy/settings.yml'
+end
+SETTINGS         = YAML.load_file(settings_file)
+PUPPETCA         = YAML.load_file(SETTINGS[:settings_directory] + '/puppetca.yml')
+protocol         = PUPPETCA[:enabled] == true ? 'https' : PUPPETCA[:enabled]
```

Comment:
Probably good if this also handles the value `false` and print a proper error

Reply:
Added.